hera_pspec/pspecdata.py

```diff
@@ -1621,6 +1686,208 @@ def get_unnormed_V(self, key1, key2, model='empirical', exact_norm=False, pol =
+            in 'dsets' mode, error bars are estimated from user-provided
+            per baseline and per channel standard deivations. If 'empirical' is
+            set, then error bars are estimated from the data by calculating the
+            channel-channel covariance of each baseline over time and
+            then applying the appropriate linear transformations to these
+            frequency-domain covariances. If 'autos' is set, the covariances of the input data
+            over a baseline is estimated from the autocorrelations of the two antennas over channel bandwidth 
+            and integration time. 
+
+        known_cov : dicts of covariance matrices
+            Covariance matrices that are not calculated internally from data.
+
+        Returns
+        -------
+        V : array_like, complex
+            Bandpower covariance, with dimension (Ntimes, spw_Ndlys, spw_Ndlys).
+        """
+        # Collect all the relevant pieces
+        
+        # Get E matrices and input covariance matrices
+        E_matrices = self.get_unnormed_E(key1, key2, exact_norm=exact_norm, pol=pol)       
+        # (spw_Ndlys, spw_Nfreqs, spw_Nfreqs)
+
+        C11, C22, C21, C12, P11, S11, P22, S22, P21, S21 = [], [], [], [], [], [], [], [], [], []
+        for time_index in range(self.dsets[0].Ntimes):
+            C11.append(self.C_model(key1, model=model, known_cov=known_cov, time_index=time_index)[np.newaxis,:,:])
+            C22.append(self.C_model(key2, model=model, known_cov=known_cov, time_index=time_index)[np.newaxis,:,:])
+            C21.append(self.cross_covar_model(key2, key1, model=model, conj_1=False, conj_2=True, known_cov=known_cov, time_index=time_index)[np.newaxis,:,:])
+            C12.append(self.cross_covar_model(key1, key2, model=model, conj_1=False, conj_2=True, known_cov=known_cov, time_index=time_index)[np.newaxis,:,:])
+            P11.append(self.cross_covar_model(key1, key1, model=model, conj_1=False, conj_2=False, known_cov=known_cov, time_index=time_index)[np.newaxis,:,:])
+            S11.append(self.cross_covar_model(key1, key1, model=model, conj_1=True, conj_2=True, known_cov=known_cov, time_index=time_index)[np.newaxis,:,:])
+            P22.append(self.cross_covar_model(key2, key2, model=model, conj_1=False, conj_2=False, known_cov=known_cov, time_index=time_index)[np.newaxis,:,:])
+            S22.append(self.cross_covar_model(key2, key2, model=model, conj_1=True, conj_2=True, known_cov=known_cov, time_index=time_index)[np.newaxis,:,:])
+            P21.append(self.cross_covar_model(key2, key1, model=model, conj_1=False, conj_2=False, known_cov=known_cov, time_index=time_index)[np.newaxis,:,:])
+            S21.append(self.cross_covar_model(key2, key1, model=model, conj_1=True, conj_2=True, known_cov=known_cov, time_index=time_index)[np.newaxis,:,:])
+        # (Ntimes, 1, spw_Nfreqs, spw_Nfreqs)
+        
+        if (C11[0] == C11[-1]).all():
```

Comment:
not good to use `==` for comparing data that is floating point--unless you expect it to be *exactly* the same. try using `np.isclose(a, b, atol=xx).all()` instead

Reply:
Sure.